src/tools/data/datagen/CMakeLists.txt

```diff
@@ -298,44 +217,64 @@ FOREACH(VTK_TARGET ${VTK_TARGETS})
         BUILD_DATA_TARGET(${VTK_TARGET}
             LIBS VTK::IOLegacy)
     endif()
-ENDFOREACH(VTK_TARGET ${VTK_TARGETS})
+endforeach()
 
 #-----------------------------------------------------------------------------
 # If we have HDF5 build all HDF5 Data Gen Target Programs
 #-----------------------------------------------------------------------------
-IF(HDF5_FOUND)
-    FOREACH(HDF5_TARGET ${HDF5_TARGETS})
+if(HDF5_FOUND)
+    foreach(HDF5_TARGET unic xdmf xdmf2d)
         BUILD_DATA_TARGET(${HDF5_TARGET}
             INCLUDE_DIRS ${CMAKE_CURRENT_SOURCE_DIR} ${HDF5_INCLUDE_DIR}
             LIB_DIRS ${HDF5_LIBRARY_DIR}
             LIBS ${HDF5_LIB})
-    ENDFOREACH()
-ENDIF(HDF5_FOUND)
+    endforeach()
+endif()
 
 #-----------------------------------------------------------------------------
 # If we have Silo build all Silo & PDB Data Gen Programs
 #-----------------------------------------------------------------------------
-IF(SILO_FOUND)
+if(SILO_FOUND)
+    #-------------------------------------------------------------------------
+    #  Setup {TARGET}_SOURCES for targets with multiple source files.
+    #-------------------------------------------------------------------------
+    set(correlationtest_SOURCES correlationtest.C pdbdatabase_link.C)
+    set(pdbdatabase_SOURCES pdbdatabase.C pdbdatabase_link.C)
+    set(streak_SOURCES streak.C pdbdatabase_link.C)
     # Build PDB Targets (The required pdb symbols live in Silo's libs.)
-    FOREACH(PDB_TARGET ${PDB_TARGETS})
+    foreach(PDB_TARGET correlationtest pdbdatabase streak)
         BUILD_DATA_TARGET(${PDB_TARGET}
             INCLUDE_DIRS ${CMAKE_CURRENT_SOURCE_DIR}
-                ${VISIT_BINARY_DIR}/include ${VISIT_SOURCE_DIR}/include
+                         ${VISIT_BINARY_DIR}/include
+                         ${VISIT_SOURCE_DIR}/include
+                         ${VISIT_SOURCE_DIR}/common/utility
+                         ${SILO_INCLUDE_DIR}
             LIB_DIRS ${PDB_LIBRARY_DIR}
             LIBS ${PDB_LIB})
-    ENDFOREACH()
+    endforeach()
+    #-------------------------------------------------------------------------
+    #  Setup {TARGET}_SOURCES for targets with multiple source files.
+    #-------------------------------------------------------------------------
```

Comment:
Am a tad confused by this block comment followed by single line `Build Silo Targets` comment. Is the block comment explaining just lines 258 and 259 or everything to 270? I think the former and if so, seems clearer to make just a single line comment. Likewise for similar block comment above about PDB targets.

Reply:
The comments were meant to justify the creation of the XXX_SOURCES variables, hence me moving it to where I moved the var creation.
I thought about modifying the BUILD_DATA_TARGET function so that the SOURCES variable could just be passed in as an argument. Would require an extra call to that function for the silo multiple source targets. I guess that wouldn't be too bad, and might make the logic cleaner. What do you think?

Reply:
Ah, ok. I think that is fine then.